pinot-controller/src/main/java/com/linkedin/pinot/controller/api/upload/ZKOperator.java

```diff
@@ -178,18 +191,20 @@ private void checkCRC(HttpHeaders headers, String offlineTableName, String segme
   }
 
   private void processNewSegment(SegmentMetadata segmentMetadata, URI finalSegmentLocationURI,
-      File currentSegmentLocation, String zkDownloadURI, String rawTableName, String segmentName) {
+      File currentSegmentLocation, String zkDownloadURI, String crypter, String rawTableName, String segmentName) {
     try {
       moveSegmentToPermanentDirectory(currentSegmentLocation, finalSegmentLocationURI);
-      LOGGER.info("Moved segment {} from temp location {} to {}", segmentName, currentSegmentLocation.getAbsolutePath(), finalSegmentLocationURI.getPath());
```

Comment:
I think you need to change moveSegmentToPermanentDirectory to move the encrypted segment instead

Reply:
The currentSegmentLocation passed into moveSegmentToPermanentDirectory is the encrypted segment now (as part of this PR).